packages/fast-usdc/src/exos/status-manager.js

```diff
@@ -122,49 +171,81 @@ export const prepareStatusManager = zone => {
       },
 
       /**
-       * Find an `ADVANCED` or `OBSERVED` tx waiting to be `SETTLED`
+       * @param {CctpTxEvidence} evidence
+       */
+      isSeen(evidence) {
+        const seenKey = seenTxKeyOf(evidence);
+        return seenTxs.has(seenKey);
+      },
+
+      /**
+       * Remove and return an `ADVANCED` or `OBSERVED` tx waiting to be `SETTLED`.
        *
        * @param {NobleAddress} address
        * @param {bigint} amount
-       * @returns {boolean}
+       * @returns {Pick<PendingTx, 'status' | 'txHash'> | undefined} undefined if nothing is settleable
        */
-      hasPendingSettlement(address, amount) {
+      dequeueStatus(address, amount) {
         const key = makePendingTxKey(address, amount);
+        if (!pendingTxs.has(key)) return undefined;
         const pending = pendingTxs.get(key);
-        return !!pending.length;
+        if (!pending.length) return undefined;
+
+        const dequeueIdx = pending.findIndex(
+          x => x.status !== PendingTxStatus.Advancing,
+        );
+        if (dequeueIdx < 0) return undefined;
+
+        if (pending.length > 1) {
+          const pendingCopy = [...pending];
+          pendingCopy.splice(dequeueIdx, 1);
+          pendingTxs.set(key, harden(pendingCopy));
+        } else {
+          pendingTxs.delete(key);
+        }
+
+        const { status, txHash } = pending[dequeueIdx];
+        // TODO: store txHash -> evidence for txs pending settlement?
+        return harden({ status, txHash });
       },
 
       /**
-       * Mark an `ADVANCED` or `OBSERVED` transaction as `SETTLED` and remove it
+       * Mark a transaction as `DISBURSED`
        *
+       * @param {EvmHash} txHash
        * @param {NobleAddress} address
        * @param {bigint} amount
        */
-      settle(address, amount) {
-        const key = makePendingTxKey(address, amount);
-        const pending = pendingTxs.get(key);
-
-        if (!pending.length) {
-          throw makeError(`No unsettled entry for ${q(key)}`);
-        }
+      disbursed(txHash, address, amount) {
+        // TODO: store txHash -> evidence for txs pending settlement?
+        console.log('TODO: vstorage update', { txHash, address, amount });
+      },
 
-        const pendingCopy = [...pending];
-        pendingCopy.shift();
-        // TODO, vstorage update for `TxStatus.Settled`
-        pendingTxs.set(key, harden(pendingCopy));
+      /**
+       * Mark a transaction as `FORWARDED`
+       *
+       * @param {EvmHash | undefined} txHash - undefined in case mint before observed
+       * @param {NobleAddress} address
+       * @param {bigint} amount
+       */
+      forwarded(txHash, address, amount) {
+        // TODO: store txHash -> evidence for txs pending settlement?
+        console.log('TODO: vstorage update', { txHash, address, amount });
       },
 
       /**
        * Lookup all pending entries for a given address and amount
        *
+       * XXX only used in tests. should we remove?
```

Comment:
yes. tests need only cover what can be called in production.

Reply:
I wrote this comment - some tests would need to be reworked if we were to remove this. They use this method to confirm state transitions.

Reply:
I wonder if they need to do so explicitly. Kind of testing the implementation.

Reply:
Maybe a better approach is to define the stores as optional initState arguments so we can access the contents in tests.

Reply:
that's even more of testing the implementation